### PR TITLE
CI: Fix jedi deprecation warning for 0.17.0 on IPython

### LINKED ISSUE #33567: CI : jedi deprecation warnings fails CI
```
ip = <IPython.core.interactiveshell.InteractiveShell object at 0x7f10269bd910>

    @async_mark()
    async def test_tab_complete_warning(self, ip):
        # GH 16409
        pytest.importorskip("IPython", minversion="6.0.0")
        from IPython.core.completer import provisionalcompleter
    
        code = "import pandas as pd; df = pd.DataFrame()"
        await ip.run_code(code)
        with tm.assert_produces_warning(None):
            with provisionalcompleter("ignore"):
>               list(ip.Completer.completions("df.", 1))

pandas/tests/frame/test_api.py:534: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <contextlib._GeneratorContextManager object at 0x7f10264bc190>
type = None, value = None, traceback = None

    def __exit__(self, type, value, traceback):
        if type is None:
            try:
>               next(self.gen)
E               AssertionError: Caused unexpected warning(s): [('DeprecationWarning', DeprecationWarning('Providing the line is now done in the functions themselves like `Script(...).complete(line, column)`'), '/home/vsts/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/jedi/api/__init__.py', 843), ('DeprecationWarning', DeprecationWarning('Providing the column is now done in the functions themselves like `Script(...).complete(line, column)`'), '/home/vsts/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/jedi/api/__init__.py', 843), ('DeprecationWarning', DeprecationWarning('Deprecated since version 0.16.0. Use Script(...).complete instead.'), '/home/vsts/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/IPython/core/completer.py', 1401)]
```

### PR DESCRIPTION
**Pull request by charlesdong1991**
- [ ] closes #33567 
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by charlesdong1991** [comment]
similar to #31323

File: pandas/tests/resample/test_resampler_grouper.py [hunk 1]
```diff
@@ -28,7 +28,17 @@ async def test_tab_complete_ipython6_warning(ip):
     )
     await ip.run_code(code)
 
-    with tm.assert_produces_warning(None):
+    # TODO: remove it when Ipython updates
```
Comment (jbrockmendel) [inline pandas/tests/resample/test_resampler_grouper.py:31]: can you add the GH ref here.

**Comment by jbrockmendel** [comment]
ex small comment, LGTM pending green

File: pandas/tests/resample/test_resampler_grouper.py (hunk 1 above)
Comment (charlesdong1991) [reply pandas/tests/resample/test_resampler_grouper.py:31]: sure, added!

**Comment by jbrockmendel** [comment]
Thanks @charlesdong1991

**Comment by simonjayhawkins** [comment]
@Meeseeksdev backport to 1.0.x

**Pull request merged**